src/pal/inc/pal_endian.h

```diff
@@ -98,8 +98,9 @@ inline void SwapGuid(GUID *pGuid)
 #define VALPTR(x) VAL32(x)
 #endif
 
-#if defined(ALIGN_ACCESS) && !defined(_MSC_VER)
-
+#if defined(_TARGET_ARM_) && !defined(_MSC_VER)
```

Comment:
@janvorli , @jkotas  Could give me a hint about the `defined(ALIGN_ACCESS)` ?  Is `ALIGN_ACCESS` coming from ./src/inc/stdmacros.h#L131 ?  I could not know the related history why this statement was created in the past, because it was made by 'dotnet-bot'. 

```
^ef1e2ab (dotnet-bot        2015-01-30 14:14:42 -0800 103 #if defined(ALIGN_ACCESS) && !def
^ef1e2ab (dotnet-bot        2015-01-30 14:14:42 -0800 104) #ifdef __cplusplus
^ef1e2ab (dotnet-bot        2015-01-30 14:14:42 -0800 105) extern "C++" {
```

When I checked a real `ALIGN_ACCESS` definition at build time, Actually, the `ALIGN_ACCESS` (in ./src/pal/inc/pal_endian.h) macro was not defined in case of cross-compilation of CoreCLR for Linux/ARM. So, I have modified the existing `if defined(ALIGN_ACCESS) && !defined(_MSC_VER)` statement to aware aggressive optimization levels (e.g., O2/O3) in Linux/ARM environment.  Could you  review the related lines that I modified?